### PR TITLE
pkg/trace/{agent,writer}: simplify and reduce channel dupes

### PR DESCRIPTION
**Pull request by gbbr**

### CODE REVIEW

File: pkg/trace/agent/agent.go [hunk 1]
```diff
@@ -255,12 +250,12 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 			ss.Size += pb.Trace(events).Msgsize()
 		}
 		if ss.Size > writer.MaxPayloadSize {
-			a.Out <- ss
+			a.TraceWriter.In <- ss
```
Comment (gbbr) [inline pkg/trace/agent/agent.go:253]: One concern with this approach is that someone external could be reading from the channel, messing up the flow. A solution to that would be to make the channel unexported and have a method like:
```go
func (*TraceWriter) In() chan<- *SampledSpans
```
Then, the returned channel can only be sent to...

Happy to make this change.

File: pkg/trace/writer/stats.go [hunk 2]
```diff
@@ -36,22 +36,26 @@ const (
...
 	easylog *logutil.ThrottledLogger
 }
 
 // NewStatsWriter returns a new StatsWriter. It must be started using Run.
-func NewStatsWriter(cfg *config.AgentConfig, inBuckets <-chan []stats.Bucket, inPayloads chan *stats.Payload) *StatsWriter {
+func NewStatsWriter(cfg *config.AgentConfig, inBuckets chan []stats.Bucket) *StatsWriter {
 	sw := &StatsWriter{
-		in:        inBuckets,
-		inPayload: inPayloads,
+		InBuckets: inBuckets,
```
Comment (bmermet) [inline pkg/trace/writer/stats.go:57]: For consistency shouldn't inBuckets be created here too?

Comment (gbbr) [reply pkg/trace/writer/stats.go:57]: The problem with it is that it's shared with the concentrator here: https://github.com/DataDog/datadog-agent/blob/a4ccdcf1d97fe803e421f803a6646092a2a2e5a9/pkg/trace/agent/agent.go#L73-L82

**Comment by gbbr** [comment]
Realised based on feedback that this could add difficulties when rebasing on `master`, so closing and eventually applying there later.

**Pull request closed without merging**